server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -5664,8 +5647,8 @@ public void testRecoverFromLocalTranslog() throws Exception {
                 for (Engine.Operation op : operations) {
                     applyOperation(engine, op);
                     if (randomBoolean()) {
+                        globalCheckpoint.set(randomLongBetween(globalCheckpoint.get(), engine.getProcessedLocalCheckpoint()));
```

Comment:
I wonder why this change is necessary? I find it a bit counter intuitive to update that before sync'ing translog, but I am sure there is a good reason.

Reply:
I've reverted this in https://github.com/elastic/elasticsearch/pull/51905/commits/630da78ad629bf1a9fb7af78c9dcee77cb7c2b66.